examples/esplora_backend_with_tor.rs

```diff
@@ -0,0 +1,126 @@
+use std::fs::OpenOptions;
+use std::io::Read;
+use std::str::FromStr;
+use std::time::Duration;
+use std::{env, thread};
+
+use bitcoin::{util::bip32, Network};
+use esplora_client::Builder;
+use libtor::{LogDestination, LogLevel, Tor, TorFlag};
+
+use bdk::{
+    blockchain::esplora::EsploraBlockchain, database::MemoryDatabase, template::Bip84,
+    KeychainKind, SyncOptions, Wallet,
+};
+/// This will create a wallet from an xpriv and sync it by connecting to an Esplora server
+/// over Tor network, using blocking calls with `ureq`.
+///
+/// This can be run with `cargo run --features="use-esplora-blocking" --example esplora_backend_with_tor`
+/// in the root folder.
+fn main() {
+    let network = Network::Testnet;
+
+    let xpriv_str =
+        "tprv8ZgxMBicQKsPcx5nBGsR63Pe8KnRUqmbJNENAfGftF3yuXoMMoVJJcYeUw5eVkm9WBPjWYt6HMWYJNesB5HaNVBaFc1M6dRjWSYnmewUMYy";
+
+    let esplora_url =
+        "http://explorerzydxu5ecjrkwceayqybizmpjjznk5izmitf2modhcusuqlid.onion/testnet/api";
+
+    let xpriv = bip32::ExtendedPrivKey::from_str(xpriv_str).unwrap();
+
+    let socks_addr = start_tor();
+
+    let client = Builder::new(esplora_url)
+        .proxy(&format!("socks5://{}", socks_addr))
+        .timeout(120) // seconds
+        .build_blocking()
+        .expect("Should never fail with no proxy and timeout");
+
+    println!(
+        "Connecting to {} via SOCKS5 proxy {}",
+        &esplora_url, &socks_addr
+    );
+
+    let blockchain = EsploraBlockchain::from_client(client, 20);
+
+    let wallet = Wallet::new(
+        Bip84(xpriv, KeychainKind::External),
+        Some(Bip84(xpriv, KeychainKind::Internal)),
+        network,
+        MemoryDatabase::default(),
+    )
+    .unwrap();
+
+    println!("Syncing the wallet");
```

Comment:
Where did the transaction creation part go? That should be included. I was only talking about printing the wallet export as unnecessary..

Reply:
As per the issue `how to use integrated rust tor client to sync bdk wallet` so there's no need in  the transaction creation part at all.

Reply:
I don't think so.. The sync doesn't make any sense if there's no wallet data to sync.. You are just creating an empty wallet and calling sync.. It doesn't complete in theory until you get to see an updated balance..